atlp-rwanda/atlp-devpulse-bn
# Edit, Publish and unpublish Job post

PR Description
This PR implements functionalities for editing, publishing, and unpublishing job posts. These updates allow administrators to efficiently manage job listings, ensuring that only relevant and up-to-date information is displayed to applicants.

Description of tasks that were expected to be completed
Edit Job Posts:

Added an editing feature for administrators to modify existing job details.

Developed a feature enabling administrators to publish job posts, making them live and accessible to applicants.

Added a confirmation prompt before publishing to prevent accidental actions.
** Unpublish Job Posts**

Created an option to unpublish job posts, allowing temporary removal from public view without deletion.

Updated backend to support running seeders, allowing the addition of necessary data for the job post functionalities.

**How has this been tested?**

1. Clone the repo and check out this branch.
2. Run npm install followed by npm run dev to start the application.
3. Set up environment variables as indicated in .env.example.
4. Run backend seeders to populate the database with necessary data.
5. use apollo server on http://localhost:4000/
6. login with email: [admin@example.com](mailto:admin@example.com), password: password123.
7. Verify that editing, publishing, and unpublishing job posts function as expected, with confirmation prompts appearing correctly.
8. 

Issue Link: https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/155

Deployed Link
N/A

Track PR (Issue number & link)
N/A

Screenshots (If appropriate)
N/A
